src/c++/perf_analyzer/inference_profiler.cc

```diff
@@ -654,38 +654,37 @@ InferenceProfiler::ProfileHelper(
 }
 
 bool
-InferenceProfiler::reached_stability(LoadStatus& load_status, bool* is_stable)
+InferenceProfiler::finished_profiling(LoadStatus& load_status, bool* is_stable)
 {
-  bool should_break = false;
+  bool done = false;
   if (load_status.infer_per_sec.size() >= load_parameters_.stability_window) {
     size_t idx =
         load_status.infer_per_sec.size() - load_parameters_.stability_window;
     *is_stable = true;
     bool within_threshold = false;
 
-    for (; idx < load_status.infer_per_sec.size(); idx++) {
-      if (load_status.infer_per_sec[idx] == 0) {
+    for (size_t i = idx; i < load_status.infer_per_sec.size(); i++) {
+      if (load_status.infer_per_sec[i] == 0) {
         *is_stable = false;
-        break;
+        return done;
```

Comment:
Is this logically equivalent to the old code?  Couldn't we in theory return true later on in the code from the within_threshold code?  

Along the same lines, I don't think a break is correct either.